### PR TITLE
Added auto-evolving.

### PR DESCRIPTION
**Pull request by Javapt**
Required candy is read from new file 'pokemon-candy.csv'
Added new configuration option which accepts a list of a pokemon to auto evolve.

### CODE REVIEW

**Comment by Lenovotro** [comment]
Looking forward to this! All these candies saved up ;)

**Comment by iamthasasquatch** [comment]
I am getting an error with the evolve.kt

e: C:\pogobot\src\main\kotlin\ink\abb\pogo\scraper\tasks\EvolvePokemon.kt: (26,
38): Unresolved reference: pokebank
e: C:\pogobot\src\main\kotlin\ink\abb\pogo\scraper\tasks\EvolvePokemon.kt: (26,
66): Unresolved reference: it

**Comment by Zeift** [comment]
Hopefully they could include this.

**Comment by Lenovotro** [comment]
Couldn't get this to work. No command prompt error, but no evolving occurring.

**Comment by Javapt** [comment]
Resolved conflicts.

**Comment by Lenovotro** [comment]
Is there a text prompt to show evolving occurring?

**Comment by Javapt** [comment]
IF you added the line to the configuration then yes you should see a message when it is going to evolve. Make sure you have this line in your properties file: 

auto_evolve=PIDGEY,RATTATA,CATERPIE,WEEDLE,DODUO,ZUBAT

Just replace those pokemon with whatever you want.

**Comment by Lenovotro** [comment]
I'll try it again. Not sure what i'm missing. Could auto-transfer disrupt any evolves from happening?

**Comment by iamthasasquatch** [comment]
23 Jul 10:54:55: Evolving PIDGEY because we have 14 candy and only need 12.

is printed when evolving.

**Comment by julienkessels** [comment]
please add this to the repo

File: src/main/kotlin/ink/abb/pogo/scraper/Bot.kt [hunk 1]
```diff
@@ -71,6 +72,7 @@ class Bot(val api: PokemonGo, val settings: Settings) {
                 if (!settings.walkOnly) {
                     task(catch)
                     task(drop)
+                    task(evolve)
```
Comment (pr0ves) [inline src/main/kotlin/ink/abb/pogo/scraper/Bot.kt]: Here should be: if autoEvolve is not empty run task(evolve), otherwise it will run an empty thread.

**Comment by falko100** [comment]
:+1:

**Comment by Lenovotro** [comment]
Works Great. thank you

**Comment by julienkessels** [comment]
how can i use your pull request already? do i need to recompile something or can i just change something in my existing folder ?

**Comment by jabbink** [comment]
>how can i use your pull request already?

Clone this, then

```git checkout -b Javapt-master master
git pull git://github.com/Javapt/PokemonGoBot.git master```

Also not merging in it's current state as I'd rather have the candy needed in a KV-map in the code (by PokemonId or sth) and add an option to wait with evolving and use a lucky egg just before.

**Comment by Forty-Tw0** [comment]
Should also be the option to only evolve the 1st pokemon in the evolutionary line to maximise XP per candy.

Could also have it evolve up the line if an evolution is not yet owned, and an option to only evolve if the resulting CP would be greater than an already owned evolved pokemon. I think this might be useful. http://pokemongo.gamepress.gg/power-up-costs

**Comment by pr0ves** [comment]
Closing this pr because wants merge with `master` branch.
Open another one asking to merge with the `develop` branch.

**Pull request closed without merging**